addons/point_of_sale/models/pos_order.py

```diff
@@ -1042,6 +1124,55 @@ def refund(self):
             'target': 'current',
         }
 
+    @api.model
+    def remove_from_ui(self, server_ids):
+        """ Remove orders from the frontend PoS application
+
+        Remove orders from the server by id.
+        :param server_ids: list of the id's of orders to remove from the server.
+        :type server_ids: list.
+        :returns: list -- list of db-ids for the removed orders.
+        """
+        orders = self.search([('id', 'in', server_ids),('state', '=', 'draft')])
+        for o in orders:
```

Comment:
you can just do:
orders.write({'state': 'cancel'})
orders.sudo().with_context(force_company=o.company_id.id).unlink()

I'm not sure of what 'orders.read(['id'])' is supposed to return as we've just deleted the records we are trying to read.
Maybe we just want to return orders.ids which are the ids we've unlinked.